lintcheck/src/json.rs

```diff
@@ -11,6 +11,7 @@ struct LintJson {
     lint: String,
     file_name: String,
     byte_pos: (u32, u32),
```

Comment:
Could remove these now since the `file_link` would cover it well enough

Reply:
The byte position is still used in the `key()` function on line 19. We could replace it with the link, but I think having it separately would make more sense. While all links currently contain line labels, there is no guarantee that they will for manually configured lints. And links also don't contain the column, which is included in this field.

Reply:
Not critical though, either way it's ready for r+ after ditching the test commit

Reply:
> Removing information from the key has a benefit - it would mean tweaks to a span's start column/end line/end column show as a change rather than

The tweaked span would still result in a `added` and `removed` message, since the lint message displays the span and is also used as part of the span. I think here it's safer to display them separately and leave it to the reviewer to merge them.

Reply:
The message isn't part of the key